### DISCUSSION TITLE
How to set ICACHE and DCACHE for ESP32-S3 in arduino-esp32?

### DISCUSSION THREAD

**Discussion by planevina** [espressif/arduino-esp32#7513] [Q&A] [answered]
I know esp-idf can do it ,user menuconfig ，set the INSTRUCTION_CACHE to 32kb and set the DATA_CACHE to 64kb
My question is :  Is there anyway to set it in arduino-esp32?
Thanks

**Comment by me-no-dev** [comment, marked as answer]
Arduino uses precompiled ESP-IDF, so in order to change the default config, you will either need to compile the libraries yourself or use Arduino as IDF component to compile your code that way.

**Reply by planevina** [reply]
ok ,thanks
